lib/chat/bloc/provider/ChatProvider.dart

```diff
@@ -280,22 +271,62 @@ class ChatProvider implements ViewModelProvider<ChatViewModel> {
     );
   }
 
-  void _updateAvatarVisibility() {
+  void _updatePreviousMessages() {
     List<MessageBubbleViewModel> messageViewModels =
         _chatViewModel.messageViewModels;
     if (messageViewModels.length >= _Constants.minMessagesLengthToUpdate) {
       MessageBubbleViewModel currentViewModel =
           messageViewModels[_Constants.currentMessageIndex];
       MessageBubbleViewModel previousViewModel =
           messageViewModels[_Constants.previousMessageIndex];
-      if (currentViewModel.messageType == MessageType.received &&
-          previousViewModel.messageType == MessageType.received) {
-        previousViewModel.avatar = _buildDefaultAvatarEmptyBox();
-      }
+      _updateReceivedMessages(
+        currentViewModel: currentViewModel,
+        previousViewModel: previousViewModel,
+      );
+    }
+  }
+
+  void _updateReceivedMessages({
+    MessageBubbleViewModel currentViewModel,
+    MessageBubbleViewModel previousViewModel,
+  }) {
+    switch (previousViewModel.messageType) {
+      case MessageType.sent:
+        currentViewModel.messageType = MessageType.received;
+        break;
+      case MessageType.received:
+        _updateFromReceived(
+          currentViewModel: currentViewModel,
+          previousViewModel: previousViewModel,
+        );
+        break;
+      case MessageType.receivedStackBottom:
```

Comment:
What a `receivedStackBottom` means?

Reply:
Basically there are 4 received message styles.

- **received**: which is a usual style.
- **receivedStackTop**: that means that we have more than one received message followed by the same type. In this case is TOP styled one.
- **receivedStackBottom**: the very last on received styles, just before the sent message appears.
- **receivedStackBetween**: if we have more than 2 received messages in column (I called it stack), this one is what we are seeing in the middle (between top and bottom)

Are you agree with this naming @albertmiro ?

Reply:
Ah ok, is for the different message styles we need for chat messages according to the design. I didn't remember that we have these three different styles. 👍